x-pack/plugins/osquery/common/types/osquery_action.ts

```diff
@@ -8,19 +8,10 @@
 export interface LogsOsqueryAction {
   '@timestamp': string;
   action_id: string;
-  alert_ids: string[];
+  alert_ids: string | string[];
+  agents: string | string[];
```

Comment:
This interface explains how the data fetched from Automated Response Actions look like. This is the very same fetch for Osquery and Endpoint actions. It says single | array because while we have 1 value in array I change the array to a single string. I agree it is not perfect, or even may be confusing, but in this what it is :P The reason for having string | array is that some would be hard to be converted on the later stage eg. EndpointAction.data.command shouldn't be an array of a single command. 

What do you think about this @ashokaditya  ? Hope this makes sense